### PR TITLE
Feature/decryption helper

### PR DESCRIPTION
**Pull request by chrisjensen**
**🤓 What should we check?**
  - Verify that private posts decrypt correctly in feeds with parallel batch processing
  - Confirm edge cases are handled (missing keys, decryption failures)

**💫 What have you changed?**
  - Added decryptBatch utility in src/lib/encryption.ts for parallel decryption of multiple encrypted items
  - Refactored decryptPosts in src/lib/api/feed/private-posts.ts to use the new batch utility, simplifying the decryption flow
  - Added comprehensive test suite in src/lib/api/feed/__tests__/private-posts.test.ts covering edge cases li

**🎟️ Which issue does this solve?**
- This PR is in preparation for #124 

**🧪 How can this be tested or verified?**
- Run yarn test src/lib/api/feed/__tests__/private-posts.test.ts to verify the new test suite passes
- View a feed containing private posts and confirm they decrypt correctly

### CODE REVIEW

File: src/lib/api/user-keys.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 
 import {generateKeyPair} from '#/lib/encryption'
+import {logger} from '#/logger'
+import * as Toast from '#/view/com/util/Toast'
 import {getErrorCode, SpeakeasyApiCall} from './speakeasy'
 
+// Throttled toast for private key errors - only show once per 30 seconds
+const showPrivateKeyErrorToast = throttle(() => {
+  Toast.show('Unable to load private data', 'xmark')
+}, 30_000)
```
Comment (f1yn) [inline src/lib/api/user-keys.ts]: Sometimes I forget how quirky JS is until I get to these parts 😂

File: src/lib/encryption.ts [hunk 2]
```diff
@@ -450,6 +450,69 @@ export async function decryptContent(
...
+    sessionKeys.map(k => [k.sessionId, k.encryptedDek]),
+  )
+
+  const results = await Promise.all(
+    items.map(async item => {
+      const encryptedDek = sessionKeyMap.get(item.sessionId)
+      if (!encryptedDek) return null
+
+      try {
+        const dek = await decryptDEK(encryptedDek, privateKey)
```
Comment (f1yn) [inline src/lib/encryption.ts]: An important constraint to note with this is that (at least in web environments): the io benefits of crpytographic mechanisms is fleeting at best: unless you're scheduling time on a worker.

To be mindful of this, there's a propbable opererational ceiling with batches on web clients because it's not going to be truly async. This would likely surface as a laggy experience, but it's unlikely to crash unless you're handling several hundred MB of data.

**Review by f1yn — APPROVED** [review]
Approved with comment (and small nit)
